### ISSUE TITLE
Add web server technology to information block you get from the dashboard

### ISSUE THREAD

**Issue by shoulders** [virtualmin/virtualmin-gpl#849]
Current system information is as follows

|  SYSTEM INFORMATION||
|----------------------|---------------------------|
|  OS type and version |     Ubuntu Linux 22.04.4     |
|  Webmin version      |     2.111      |
|  Usermin version     |     2.010     |
|  Virtualmin version  |     7.10.0       |
|  Theme version       |     21.10      |
|  Package updates     |      94  package updates are available   |

Can you add the web technology used i.e. Apache, NGINX, Apache+NGINX

This helps out when giving support on the forum as sometimes the web server technology is assumed to be Apache but the user is using NGINX.

Suggested by **stegan** https://forum.virtualmin.com/t/limit-of-attachments-in-roundcube/127612/14

## Example

|  SYSTEM INFORMATION||
|----------------------|---------------------------|
|  OS type and version |     Ubuntu Linux 22.04.4     |
|  Webserver |     Apache 2.4.11     |
|  Webmin version      |     2.111      |
|  Usermin version     |     2.010     |
|  Virtualmin version  |     7.10.0       |
|  Theme version       |     21.10      |
|  Package updates     |      94  package updates are available   |

**Comment by iliajie** [comment]
Hello,

Thanks for the heads up! Though, I don't believe it's necessary.

**Comment by jcameron** [comment]
Actually I think this would be useful to display..

**Comment by iliajie** [comment]
There's a 99% chance it's clear what the user is talking about. In the Authentic theme, we also have the option to return full set of system information by holding the Shift key and clicking on the clipboard icon on the top right of the dashboard, which returns:


|SYSTEM INFORMATION&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| |
|---|---|
|  System hostname  |  host.rocky9-pro.virtualmin.dev (10.211.55.87)  |
|  Operating system  |  Rocky Linux 9.4  |
|  Time on system  |  Tuesday, July 2, 2024 11:14 PM  |
|  Kernel and CPU  |  Linux 5.14.0-427.22.1.el9_4.x86_64 on x86_64  |
|  Processor information  |  Intel(R) Core(TM) i7-1068NG7 CPU @ 2.30GHz, 2 cores  |
|  System uptime  |  10 hours, 37 minutes  |
|  Running processes  |  181  |
|  CPU load averages  |  1.23 (1 min) 1.26 (5 mins) 1.14 (15 mins)  |
|  Real memory  |  871.17 MiB used / 168.49 MiB cached / 1.23 GiB total  |
|  Virtual memory  |  984.5 MiB used / 1.6 GiB total  |
|  Local disk space  |  10.75 GiB used / 3.62 GiB free / 14.37 GiB total  |
|  Package updates  |  13 package updates are available  |

|SOFTWARE VERSIONS&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| |
|---|---|
|  host.rocky9-pro.virtualmin.dev (10.211.55.87)  |  Operating system  |
|  Webmin version  |  2.111.202407012133  |
|  Usermin version  |  2.010.202406300102  |
|  Virtualmin version  |  7.10.0 Pro  |
|  Authentic theme version  |  21.20-RC3-2  |
|  Path to Perl  |  /usr/bin/perl  |
|  Python version  |  3.9.18  |
|  Path to Python  |  /bin/python3.9  |
|  BIND version  |  9.16  |
|  Postfix version  |  3.5.9  |
|  Mail injection command  |  /usr/lib/sendmail -t  |
|  Apache version  |  2.4.57  |
|  PHP versions  |  7.4.33, 8.0.30, 8.1.29, 8.2.20  |
|  Logrotate version  |  3.18.0  |
|  MariaDB version  |  10.5.22  |
|  SpamAssassin version  |  3.4.6  |

|VIRTUALMIN COUNTS&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| |
|---|---|
|  Virtual servers  |  1  |
|  DNS domains  |  1  |
|  Virtual websites  |  1  |
|  SSL websites  |  1  |
|  Mail domains  |  1  |
|  Databases  |  2  |
|  Mail/FTP users  |  1  |
|  Mail aliases  |  4  |

I think the inability to provide the required details in the first place will correlate with the inability to copy/paste this information from the dashboard or clearly describe the initial problem making this whole change ineffective.

Even though we could add it to the basic version of system information, it raises a logical question: why not include PHP information or database information as well? Should we do the same for consistency in the forum post template? Joe might disagree with having too many fillable fields.

**Comment by jcameron** [comment]
I think we should just show this in the "SOFTWARE VERSIONS" section.

**Comment by iliajie** [comment]
> I think we should just show this in the "SOFTWARE VERSIONS" section.

Yes, and we already provide this in extended system information. Do you now mean we should also do it for basic system information?

You can see the difference between the basic and extended system information by clicking this button with and without holding the Shift key.

<img width="523" alt="image" src="https://github.com/virtualmin/virtualmin-gpl/assets/4426533/5e6cc350-99cd-45d7-b701-00fb3b941699">

**Comment by shoulders** [comment]
I think this should go in basic because when installing virtualmin you can select LEMP/LAMP but the rest of the technology is the same for the rest of the install.

Maybe the mode in basic as follows

(Apache | NGINX | Apache + NGINX)

There has been a lot of issues with proxying  from what I can see on the forum people getting mixed up between the 2 different technologies.

I do appreciate that the reporter should take some time to give the correct information.

The extended information is ideal when more in-depth discussion or investigation is needed but not required for general forum discussion.

**Comment by iliajie** [comment]
Alright, I have checked that code and reworked it to support extracting a given list of keys, whichever we want in the basic information. I also fixed a bug that had gone unnoticed for years, so now the full system information will be more precise.

Though it was a huge rework, a _lot_ of work! The upside is that the code is much cleaner now, and the returned markdown is immaculate.

However, I still don't believe that many users will find this useful, as practice has shown.

This is an example of how it will look now:

| SYSTEM INFORMATION ||
|----------|----------|
|  OS type and version |  Rocky Linux 9.4                  |
|  Webmin version      |  2.111.202407030032               |
|  Usermin version     |  2.010.202407030032               |
|  Virtualmin version  |  7.10.0 Pro                       |
|  Theme version       |  21.20-RC3-5                      |
|  Apache version      |  2.4.57                           |
|  Package updates     |  2 package updates are available  |

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run. @shoulders, sorry for the noise, and thanks for the contribution. We definitely didn't mean any disrespect.

If it turns out the issue is already resolved for you, you're welcome to close it.
